src/object.rs

```diff
@@ -531,6 +531,17 @@ macro_rules! glib_object_wrapper {
             }
         }
 
+        #[doc(hidden)]
+        impl $crate::translate::FromGlibPtrBorrow<*const $ffi_name> for $name {
+            #[inline]
+            #[cfg_attr(feature = "cargo-clippy", allow(cast_ptr_alignment))]
+            unsafe fn from_glib_borrow(ptr: *const $ffi_name) -> Self {
+                debug_assert!($crate::types::instance_of::<Self>(ptr as *const _));
+                $name($crate::translate::from_glib_borrow(ptr as *mut _),
+                      ::std::marker::PhantomData)
```

Comment:
And here. And the assertion is already in the function you'd call then, and could also be removed